### ISSUE TITLE
Yielded values stomp on each other from regenerator.

### ISSUE THREAD

**Issue by tolmasky** [babel/babel#2385]
If we look at the example below:

```javascript
var x = function *generator()
{
    (yield 1)(yield 2)
}

i = x()

console.log(i.next().value);
console.log(i.next(console.log).value);
console.log(i.next(3).value);
```
In node v0.12 (using real generators), the result is:

```javascript
1
2
3
undefined
```

However, under babel, the result is:

```javascript
1
2

/Users/tolmasky/.nvm/v0.10.39/lib/node_modules/babel/node_modules/babel-core/node_modules/regenerator/runtime.js:505
        throw exception;
              ^
TypeError: number is not a function
```

To take a deeper look, we can use two functions:

```javascript
var x = function *generator()
{
    (f1 = yield 1)(f2 = yield 2)
    
    console.log(f1 + " " + f2);
}

i = x()

i.next();
i.next(function f1() { });
i.next(function f2() { });
```

We'd expect:

```javascript
function f1() {} function f2() {}
```

But instead get:

```javascript
function f2() {} function f2() {}
```

The second value stomps on the old.

**Comment by tolmasky** [comment]
Just from looking at the generated code, it appears that with this type of expression a combined switch step is produced:

```javascript
var x = regeneratorRuntime.mark(function generator() {
    return regeneratorRuntime.wrap(function generator$(context$1$0) {
        while (1) switch (context$1$0.prev = context$1$0.next) {
            case 0:
                context$1$0.next = 2;
                return 1;

            case 2:
                context$1$0.next = 4;
                return 2;

            case 4:
                context$1$0.t0 = f2 = context$1$0.sent;
                (f1 = context$1$0.sent)(context$1$0.t0);

                console.log(f1 + " " + f2);

            case 7:
            case "end":
                return context$1$0.stop();
        }
    }, generator, this);
});
```

Compare to what happens if you just change it to (f1 = yield 1) + (f2 = yield 2) (another pexression with two yields):

var x = regeneratorRuntime.mark(function generator() {
    return regeneratorRuntime.wrap(function generator$(context$1$0) {
        while (1) switch (context$1$0.prev = context$1$0.next) {
            case 0:
                context$1$0.next = 2;
                return 1;

            case 2:
                context$1$0.t0 = f1 = context$1$0.sent;
                context$1$0.next = 5;
                return 2;

            case 5:
                context$1$0.t1 = f2 = context$1$0.sent;
                context$1$0.t0 + context$1$0.t1;

                console.log(f1 + " " + f2);

            case 8:
            case "end":
                return context$1$0.stop();
        }
    }, generator, this);
});
```

**Comment by sebmck** [comment]
Moving to facebook/regenerator#218.

**Comment by benjamn** [comment]
Per https://github.com/facebook/regenerator/issues/218#issuecomment-264195266, this will be fixed when #4881 is merged.
